### PR TITLE
feat: Add bandwidth stats to `PeerManagerActor`

### PR DESCRIPTION
**Pull request by pmnoxx**
We would like to track the amount of bandwidth used for all `PeerActors`.

See https://github.com/near/nearcore/issues/5155

### CODE REVIEW

File: chain/network/src/peer_manager/peer_manager_actor.rs [hunk 1]
```diff
@@ -1534,6 +1576,9 @@ impl Actor for PeerManagerActor {
 
         // Update routing table and prune edges that are no longer reachable.
         self.update_routing_table_trigger(ctx);
+
+        // Update routing table and prune edges that are no longer reachable.
```
Comment (mm-near) [inline chain/network/src/peer_manager/peer_manager_actor.rs:1580]: bad comment

File: chain/network/src/types.rs [hunk 2]
```diff
@@ -457,6 +458,8 @@ pub struct Consolidate {
     pub(crate) other_edge_info: EdgeInfo,
     // Protocol version of new peer. May be higher than ours.
     pub(crate) peer_protocol_version: ProtocolVersion,
+    /// A helper data structure for limiting reading, reporting bandwidth stats.
```
Comment (mm-near) [inline chain/network/src/types.rs:461]: nit: is it a 'data structure' ? ;-) or simply a class

File: utils/near-rate-limiter/src/framed_read.rs [hunk 3]
```diff
@@ -254,6 +277,8 @@ where
 
             let bytect = ready!(poll_read_buf(pinned.inner.as_mut(), cx, &mut state.buffer)?);
 
+            pinned.throttle_controller.report_bandwidth_used(bytect);
+
```
Comment (mm-near) [inline utils/near-rate-limiter/src/framed_read.rs:281]: can you add a test that checks that throttler.compute_msg_seen / consume_bandwidth_used is working.

**Review by mm-near — DISMISSED** [review]

File: chain/network/src/types.rs (hunk 2 above)
Comment (pmnoxx) [reply chain/network/src/types.rs:461]: @mm-near Rust doesn't support classes: https://stevedonovan.github.io/rust-gentle-intro/object-orientation.html

**Comment by pmnoxx** [comment]
@mm-near Can you take a look? This change is important in order to be able to ship this feature.

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**